### PR TITLE
[Navigation] Register all plugins to NavGroups

### PR DESCRIPTION
**Pull request by TackAdam**
### Description
This change adds a function that registers all the plugins to their corresponding NavGroups and NavCategories:
<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:////Users/tackadam/Library/Group%20Containers/UBF8T346G9.Office/TemporaryItems/msohtmlclip/clip.htm">
<link rel=File-List
href="file:////Users/tackadam/Library/Group%20Containers/UBF8T346G9.Office/TemporaryItems/msohtmlclip/clip_filelist.xml">

</head>

<body link="#0563C1" vlink="#954F72">

Plug-in | Group | Category
-- | -- | --
Reports | Observability | Dashboards and report
Reports | Security Analytics | Dashboards and report
Reports | Search | Analyze search

</body>

</html>
Screenshot 2024-06-27 at 10 11 08 AM
Screenshot 2024-06-27 at 10 21 27 AM
Screenshot 2024-06-27 at 10 21 41 AM
Screenshot 2024-06-27 at 10 21 55 AM

### Issues Resolved
https://github.com/opensearch-project/OpenSearch-Dashboards/issues/7029

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [X] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: public/plugin_nav.tsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import { CoreSetup } from '../../../src/core/public';
+import { ReportsDashboardsPluginSetup } from './types';
+import { PLUGIN_ID } from '../common';
+import { DEFAULT_NAV_GROUPS, DEFAULT_APP_CATEGORIES } from '../../../src/core/public';
+
+export function registerAllPluginNavGroups(core: CoreSetup<ReportsDashboardsPluginSetup>) {
+  core.chrome.navGroup.addNavLinksToGroup(DEFAULT_NAV_GROUPS.observability, [
+    {
+      id: PLUGIN_ID,
+      category: DEFAULT_APP_CATEGORIES.dashboardAndReport,
```
Comment (SuZhou-Joe) [inline public/plugin_nav.tsx:15]:
```suggestion
      category: DEFAULT_APP_CATEGORIES.visualizeAndReport,
      order: 300,
```

File: public/plugin_nav.tsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    },
+  ]);
+  core.chrome.navGroup.addNavLinksToGroup(DEFAULT_NAV_GROUPS[`security-analytics`], [
+    {
+      id: PLUGIN_ID,
+      category: DEFAULT_APP_CATEGORIES.dashboardAndReport,
```
Comment (SuZhou-Joe) [inline public/plugin_nav.tsx:21]:
```suggestion
      category: DEFAULT_APP_CATEGORIES.visualizeAndReport,
      order: 300,
```

File: public/plugin_nav.tsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    },
+  ]);
+  core.chrome.navGroup.addNavLinksToGroup(DEFAULT_NAV_GROUPS.analytics, [
+    {
+      id: PLUGIN_ID,
+      category: DEFAULT_APP_CATEGORIES.dashboardAndReport,
+      showInAllNavGroup: true,
+    },
+  ]);
```
Comment (SuZhou-Joe) [inline public/plugin_nav.tsx:30]:
```suggestion
```
Reports should not be registered to analytics.

File: public/plugin_nav.tsx [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  core.chrome.navGroup.addNavLinksToGroup(DEFAULT_NAV_GROUPS.search, [
+    {
+      id: PLUGIN_ID,
+      category: DEFAULT_APP_CATEGORIES.analyzeSearch,
```
Comment (SuZhou-Joe) [inline public/plugin_nav.tsx:34]:
```suggestion
      category: DEFAULT_APP_CATEGORIES.analyzeSearch,
      order: 300,
```

**Review by SuZhou-Joe — APPROVED** [review]

**Review by rupal-bq — APPROVED** [review]

File: public/plugin_nav.tsx [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+export function registerAllPluginNavGroups(core: CoreSetup<ReportsDashboardsPluginSetup>) {
+  core.chrome.navGroup.addNavLinksToGroup(DEFAULT_NAV_GROUPS.observability, [
+    {
+      id: PLUGIN_ID,
+      category: DEFAULT_APP_CATEGORIES.visualizeAndReport,
+      order: 300,
+    },
+  ]);
+  core.chrome.navGroup.addNavLinksToGroup(DEFAULT_NAV_GROUPS[`security-analytics`], [
```
Comment (derek-ho) [inline public/plugin_nav.tsx:19]: Is there something preventing us from accessing this in the same way as the others?

File: public/plugin_nav.tsx [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    {
+      id: PLUGIN_ID,
+      category: DEFAULT_APP_CATEGORIES.visualizeAndReport,
+      order: 300,
+    },
+  ]);
+  core.chrome.navGroup.addNavLinksToGroup(DEFAULT_NAV_GROUPS.search, [
```
Comment (derek-ho) [inline public/plugin_nav.tsx:26]: Based on the description it seems like you missed the registration in the "analytics" category?

Comment (TackAdam) [reply public/plugin_nav.tsx:26]: The update removed it from analytics.

File: public/plugin_nav.tsx (hunk 5 above)
Comment (TackAdam) [reply public/plugin_nav.tsx:19]: This is the only way i am aware of is to register security-analytics in this way due to the "-" symbol

**Review by ps48 — APPROVED** [review]

**Pull request merged**